### PR TITLE
Use vars function to get attributes in order of definition

### PR DESCRIPTION
**Pull request by gowthamvbhat**

### CODE REVIEW

File: pydictable/core.py [hunk 1]
```diff
@@ -100,12 +100,16 @@ def __get_field_by_type_hint(cls, type_hint):
...
-        for attr in inspect.getmembers(cls):
-            if isinstance(attr[1], Field):
-                fields[attr[0]] = attr[1]
+        dictable_fields = {k: v for k, v in dict(vars(cls)).items() if isinstance(v, Field)}
         for name, th in get_type_hints(cls).items():
-            if name not in fields:
+            if name in dictable_fields:
+                fields[name] = dictable_fields[name]
+            else:
                 fields[name] = cls.__get_field_by_type_hint(th)
```
Comment (gowthamvbhat) [inline pydictable/core.py]: Depending on get_type_hints() method for the order, since all fields will have type hints.

File: pydictable/core.py [hunk 2]
```diff
@@ -100,12 +100,16 @@ def __get_field_by_type_hint(cls, type_hint):
...
+
+        for attr in inspect.getmembers(cls):
+            if attr[0] not in fields and isinstance(attr[1], Field):
+                fields[attr[0]] = attr[1]
```
Comment (gowthamvbhat) [inline pydictable/core.py]: This adds fields **without** type hints and also for backward compatibility

File: pydictable/core.py [hunk 3]
```diff
@@ -100,12 +100,16 @@ def __get_field_by_type_hint(cls, type_hint):
     @classmethod
     def get_fields(cls) -> Dict[str, Field]:
         fields = {}
-        for attr in inspect.getmembers(cls):
-            if isinstance(attr[1], Field):
-                fields[attr[0]] = attr[1]
+        dictable_fields = {k: v for k, v in dict(vars(cls)).items() if isinstance(v, Field)}
         for name, th in get_type_hints(cls).items():
```
Comment (pskd73) [inline pydictable/core.py:106]: Can we not change the existing implementation? Something like this
```python
fields = {}
...
ordered_fields = {}
partial_ordered_fields = {... dict(vars())}
for k, field in partial_ordered_fields:
    if k in fields:
        ordered_fields[k] = field
for k, field in fields:
    if k not in ordered_fields:
        ordered_fields[k] = field
assert set(fields.keys()) == set(ordered_fields.keys())
return ordered_fields
```

**Review by pskd73 — CHANGES_REQUESTED** [review]

File: pydictable/core.py (hunk 3 above)
Comment (gowthamvbhat) [reply pydictable/core.py:106]: Updated

File: pydictable/core.py [hunk 4]
```diff
@@ -106,7 +106,17 @@ def get_fields(cls) -> Dict[str, Field]:
...
             if name not in fields:
                 fields[name] = cls.__get_field_by_type_hint(th)
-        return fields
+
+        ordered_fields = {}
+        for name, value in dict(vars(cls)).items():
```
Comment (pskd73) [inline pydictable/core.py]: Can we only take name from this and add it to ordered fields?

**Review by pskd73 — CHANGES_REQUESTED** [review]

File: pydictable/core.py (hunk 4 above)
Comment (gowthamvbhat) [reply pydictable/core.py:111]: The vars() includes any other attributes other than FieldType and even any methods on the class.
Moved the instance check for Field is below, as it was not readable as a single line.

Comment (gowthamvbhat) [reply pydictable/core.py]: Updated

**Review by pskd73 — DISMISSED** [review]

**Review by pskd73 — APPROVED** [review]

**Pull request merged**